### PR TITLE
Sc-15455 : remove team member feature

### PR DESCRIPTION
**Pull request by masskoder**
### Scope of changes

This adds :
 - remove team member feature
 - google analytics

Fixes SC-XXXX

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

https://www.awesomescreenshot.com/video/16429957?key=2b29ef8c78849c00de76cb20a33fa7d1

### Author checklist

- [ ] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: web/beacon-app/src/features/teams/components/AddNewMember/AddNewMemberModal.tsx [hunk 1]
```diff
@@ -23,19 +23,21 @@ function AddNewMemberModal({ onClose, isOpened }: AddNewMemberModalProps) {
 
   useEffect(() => {
     if (wasMemberCreated) {
-      toast.success('Member created successfully');
+      toast.success('Success! You have invited your teammate to join your organization');
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/AddNewMember/AddNewMemberModal.tsx:26]:
```suggestion
      toast.success('Success! You have invited your teammate to join your organization.');
```

File: web/beacon-app/src/features/teams/components/ChangeRoleModal.tsx [hunk 2]
```diff
@@ -41,6 +41,10 @@ function ChangeRoleModal({ openChangeRoleModal, setOpenChangeRoleModal }: Change
             setOpenChangeRoleModal({ ...openChangeRoleModal, opened: false });
           }
         },
+        onSuccess() {
+          toast.success('Success! You have updated your teammate’s role in your organization.');
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/ChangeRoleModal.tsx:45]:
```suggestion
          toast.success('The team member’s role is updated.');
```

Comment (daniellemaxwell) [reply web/beacon-app/src/features/teams/components/ChangeRoleModal.tsx:45]: I saw this success message in the Figma file.

File: web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberForm.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+function DeleteMemberForm({ onSubmit, isSubmitting, initialValues }: NewMemberFormProps) {
+  const formik = useDeleteMemberForm(onSubmit, initialValues);
+
+  const { getFieldProps, setFieldValue, values } = formik;
+
+  return (
+    <FormikProvider value={formik}>
+      <Form className="space-y-3">
+        <TextField type="hidden" {...getFieldProps('id')} />
+        <StyledTextField label="Remove Team Member" {...getFieldProps('name')} isDisabled />
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberForm.tsx:27]:
```suggestion
        <StyledTextField label="Team Member" {...getFieldProps('name')} isDisabled />
```

File: web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberForm.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            }}
+            data-testid="delete_agreement"
+          >
+            Check to confirm removal. The team member will no longer have access to the
+            organization.
+          </Checkbox>
+        </CheckboxFieldset>
+        <div className="pt-3 text-center">
+          <Button type="submit" isLoading={isSubmitting} isDisabled={!values.delete_agreement}>
+            Delete
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberForm.tsx:42]:
```suggestion
            Remove
```

File: web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    delete_agreement: false,
+  };
+
+  const onDelete = () => {
+    deleteMember();
+  };
+
+  useEffect(() => {
+    if (wasMemberDeleted) {
+      toast.success('Success! You have removed your teammate from your organization');
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx:35]:
```suggestion
      toast.success('The team member has been removed.');
```

Comment (daniellemaxwell) [reply web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx:35]: I also saw this message in the Figma file. If the message doesn't change, should a `.` be added after `organization`?

File: web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+      onClose();
+      reset();
+    }
+  }, [wasMemberDeleted, onClose, reset]);
+
+  useEffect(() => {
+    if (hasMemberFailed) {
+      toast.error(
+        (error as any)?.response?.data?.error ||
+          `Could not delete member. Please try again or contact support,  if the problem continues.`
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx:45]:
```suggestion
          `We could not complete the action. Please try again.`
```

Comment (daniellemaxwell) [reply web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx:45]: Another message from the Figma file. If we change to this message, the error message in `AddNewMemberModal` should also be updated.

If the message doesn't change, should the space before `if the problem continues.` be deleted?

File: web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  onClose: () => void;
+};
+
+function DeleteMemberModal({ onClose, onOpen }: DeleteMemberModalProps) {
+  const { opened, member } = onOpen;
+  const { deleteMember, isDeletingMember, wasMemberDeleted, hasMemberFailed, error, reset } =
+    useDeleteMember(member?.id);
+  const initialValues = {
+    id: member?.id,
+    name: member?.name || member?.email || '---',
```
Comment (daniellemaxwell) [inline web/beacon-app/src/features/teams/components/DeleteMember/DeleteMemberModal.tsx:25]: I believe the table currently displays `-` in the `name` field for invited users. Should it be updated to `---` to match the initial value?

**Review by daniellemaxwell — APPROVED** [review]
This looks good. I was able to test the change without an issue. I've made some small suggestions to match the Figma file. Great work!

File: web/beacon-app/src/features/teams/components/ChangeRoleModal.tsx (hunk 2 above)
Comment (masskoder) [reply web/beacon-app/src/features/teams/components/ChangeRoleModal.tsx:45]: This success message is from Edwin in our last call

**Pull request merged**